src/zjs_callbacks.c

```diff
@@ -323,17 +343,33 @@ void zjs_remove_callback(zjs_callback_id id)
             }
             jerry_release_value(cb_map[id]->this);
         }
-        zjs_free(cb_map[id]);
-        cb_map[id] = NULL;
+        SET_CB_REMOVED(cb_map[id]->flags);
+        if (!skip_flush) {
+            int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id,
+                                            CB_FLUSH_ONE, NULL, 0);
+            if (ret) {
+                // couldn't add flush command, so just free now
+                zjs_flush_callback(id);
```

Comment:
Maybe print an error/debug message here because this likely means the ring buffer is full.